### PR TITLE
Implement checkpoints on show mnemonic

### PR DESCRIPTION
**Pull request by RCasatta**

### CODE REVIEW

File: firmware/src/checkpoint.rs [hunk 1]
```diff
@@ -60,6 +60,9 @@ pub enum CheckpointVariant {
 
     #[cbor(n(7))]
     Removed,
+
+    #[cbor(n(8))]
+    ShowMnemonic,
```
Comment (afilini) [inline firmware/src/checkpoint.rs:65]: I would add a field in here to save whether it's a "show mnemonic" command or an initial backup. I think we can reuse this checkpoint logic there as well, but when we reboot we need to distinguish which handle to resume

File: firmware/src/handlers/init.rs [hunk 2]
```diff
@@ -622,7 +622,23 @@ pub async fn handle_show_mnemonic(
...
+        peripherals
+            .nfc
+            .send(model::Reply::DelayedReply)
+            .await
+            .unwrap();
+    }
+
     let page = LoadingPage::new();
     page.init_display(&mut peripherals.display)?;
     page.draw_to(&mut peripherals.display)?;
```
[the remark this answers is not in the document]
Comment (RCasatta) [reply firmware/src/handlers/init.rs]: Done in 3127bd7680a918170074b0aee6056f3d69d51a8c

File: firmware/src/checkpoint.rs (hunk 1 above)
Comment (RCasatta) [reply firmware/src/checkpoint.rs]: Is a bool enough? Done in 8704eae59889020230f1fa96b0e982bd4628fb5b

**Review by afilini — APPROVED** [review]
ACK 8704eae59889020230f1fa96b0e982bd4628fb5b

**Pull request merged**